odd-platform-ui/src/components/Management/DataSourcesList/DataSourceItem/DataSourceItemToken/DataSourceItemToken.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import {
+  DataSource,
+  DataSourceApiRegenerateDataSourceTokenRequest,
+} from 'generated-sources';
+import ConfirmationDialog from 'components/shared/ConfirmationDialog/ConfirmationDialog';
+import { Typography } from '@mui/material';
+import AppButton from 'components/shared/AppButton/AppButton';
+import CopyButton from 'components/shared/CopyButton/CopyButton';
+import { TokenContainer, Token } from './DataSourceItemTokenStyles';
+
+interface DataSourceItemProps {
+  dataSource: DataSource;
+  regenerateDataSourceToken: (
+    params: DataSourceApiRegenerateDataSourceTokenRequest
+  ) => Promise<DataSource>;
+}
+
+const DataSourceItemToken: React.FC<DataSourceItemProps> = ({
+  dataSource,
+  regenerateDataSourceToken,
+}) => {
+  const checkTokenLength = (token: string): boolean =>
+    token.substring(0, 6) === '******';
+
+  const [isHidden, setIsHidden] = React.useState<boolean>(true);
+
+  React.useEffect(() => {
+    setIsHidden(checkTokenLength(dataSource.token.value));
```

Comment:
In fact I think we can remove the checkTokenLength method, since it is used in one place.
Why not just check it right here?
```suggestion
    setIsHidden(dataSource.token.value.substring(0, 6) === '******');
```